### PR TITLE
Smooth random voltages script

### PR DESCRIPTION
**Pull request by awonak**
3 smooth unique random voltages paired with matching stepped random voltages.

### CODE REVIEW

**Comment by awonak** [comment]
I've got a few issues I'd like to fix before we merge.
1. slew on short voltage hops is not very smooth (need to increase granularity of slew)
2. analog input is not outputting 1:1 (pitch cv does not retain accurate cv value on output)

**Review by mjaskula — COMMENTED** [review]
Looks good to me so far.  I'll hold off on approval since you said you are not finished yet.

File: software/contrib/smooth_random_voltages.py
Comment (awonak) [reply software/contrib/smooth_random_voltages.py]: Done. I ended up removing it as it wasn't really necessary here.

**Review by mjaskula — CHANGES_REQUESTED** [review]
Looks pretty good to me. I've opened up some discussion on a couple ways to improve the EnvelopeGenerator class, as well as a documentation change. Let me know what you think.

File: software/contrib/smooth_random_voltages.py
Comment (awonak) [reply software/contrib/smooth_random_voltages.py]: Both good calls, I went with the generator function instead of the class wrapper.

File: software/contrib/smooth_random_voltages.py
Comment (awonak) [reply software/contrib/smooth_random_voltages.py:5]: Done.

> This should especially help people who are new to synthesis and might not make the connection.

While this is helpful, I don't think we should make this a requirement for script authors.

**Review by mjaskula — APPROVED** [review]
Thanks for making these changes. The code looks good to me now. I haven't had a chance yet to fire it up and try it and see the UI, but I don't think that is required for merge. Especially since @roryjamesallen has run it. In general, it's probably a good idea if we verify that a script runs on a couple EuroPis before merge.

File: software/contrib/smooth_random_voltages.py
Comment (awonak) [reply software/contrib/smooth_random_voltages.py]: I doubled the max slew rate and added a constant to make it easier to adjust in the script.

The tricky thing with the exponential slew is that if the max slew rate is too high, then lower knob settings become much trickier to dial in versus the larger values. I originally had a linear slew which didn't have this problem, but I liked the effect of the exponential curve better, so it's a trade-off.

**Comment by awonak** [comment]
I finally got a chance to test this script on my module one more time. Worked as expected, I'm gonna merge now.

**Pull request merged**
